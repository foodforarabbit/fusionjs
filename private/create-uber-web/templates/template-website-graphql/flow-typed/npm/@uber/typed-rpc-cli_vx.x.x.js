// flow-typed signature: 8615971bf56dea74cee3836e133e434f
// flow-typed version: <<STUB>>/@uber/typed-rpc-cli_v^2.3.0/flow_v0.102.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@uber/typed-rpc-cli'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@uber/typed-rpc-cli' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@uber/typed-rpc-cli/bin/cli' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/config.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/convert-case.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/flow-from-schema' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/graphql-link' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/merge-configs' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/multiple-service-graphql' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/multiple-service-web-rpc' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/replace-extension.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/single-service-graphql' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/single-service-web-rpc' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/staging-graphql' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/update-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/add-comment' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/babel-opts' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/convert-case' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/ensure-import-declaration' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/file-utils' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-graphql-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-plugin-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-plugin-register-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-register-graphql-links' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-register-graphql-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-resolvers' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-rpc-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-schemas' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-type-imports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-atreyu-config-path' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-flow-files' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-json-descriptor-files' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-named-exports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-rpc-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-services' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-thrift-files' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-unique-token-plugin-pair' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/graphql-to-flow/index' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/graphql-to-flow/index.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/index' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/link' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/load-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/main' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/mod-flow' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/codegen-link-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/codegen-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate-graphql-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate-register-graphql-links' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate-resolvers' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate-rpc-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate-schemas' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/generate' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/get-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/get-filtered-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/get-paths' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/get-rpc-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/idl' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/index' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/install-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/link' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/lint-projects' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/merge-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/babel.config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/enzyme.config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/globals' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme_v3.x.x' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme-adapter-react-16_vx.x.x' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/jest_v23.x.x' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/src/test-utils/graphql-test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/register-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/scaffold-plugin' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/update-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/monorepo/update-link-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/parse-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/proto-to-flow/__tests__/index.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/proto-to-flow/index' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/remove-unused-imports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/replace-extension' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/test-utils/run-test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/to-cjs' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/update-flow' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/update-main-schema' {
  declare module.exports: any;
}

// Filename aliases
declare module '@uber/typed-rpc-cli/bin/cli.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/bin/cli'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/config.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/config.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/convert-case.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/convert-case.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/flow-from-schema.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/flow-from-schema'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/graphql-link.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/graphql-link'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/merge-configs.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/merge-configs'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/multiple-service-graphql.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/multiple-service-graphql'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/multiple-service-web-rpc.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/multiple-service-web-rpc'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/replace-extension.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/replace-extension.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/single-service-graphql.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/single-service-graphql'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/single-service-web-rpc.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/single-service-web-rpc'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/staging-graphql.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/staging-graphql'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/update-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/update-config'>;
}
declare module '@uber/typed-rpc-cli/src/add-comment.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/add-comment'>;
}
declare module '@uber/typed-rpc-cli/src/babel-opts.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/babel-opts'>;
}
declare module '@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake'>;
}
declare module '@uber/typed-rpc-cli/src/convert-case.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/convert-case'>;
}
declare module '@uber/typed-rpc-cli/src/ensure-import-declaration.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/ensure-import-declaration'>;
}
declare module '@uber/typed-rpc-cli/src/file-utils.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/file-utils'>;
}
declare module '@uber/typed-rpc-cli/src/generate-graphql-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-graphql-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-plugin-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-plugin-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-plugin-register-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-plugin-register-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-register-graphql-links.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-register-graphql-links'>;
}
declare module '@uber/typed-rpc-cli/src/generate-register-graphql-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-register-graphql-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/generate-resolvers.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-resolvers'>;
}
declare module '@uber/typed-rpc-cli/src/generate-rpc-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-rpc-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-schemas.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-schemas'>;
}
declare module '@uber/typed-rpc-cli/src/generate-type-imports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-type-imports'>;
}
declare module '@uber/typed-rpc-cli/src/generate.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate'>;
}
declare module '@uber/typed-rpc-cli/src/get-atreyu-config-path.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-atreyu-config-path'>;
}
declare module '@uber/typed-rpc-cli/src/get-flow-files.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-flow-files'>;
}
declare module '@uber/typed-rpc-cli/src/get-json-descriptor-files.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-json-descriptor-files'>;
}
declare module '@uber/typed-rpc-cli/src/get-named-exports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-named-exports'>;
}
declare module '@uber/typed-rpc-cli/src/get-rpc-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-rpc-code'>;
}
declare module '@uber/typed-rpc-cli/src/get-services.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-services'>;
}
declare module '@uber/typed-rpc-cli/src/get-thrift-files.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-thrift-files'>;
}
declare module '@uber/typed-rpc-cli/src/get-unique-token-plugin-pair.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-unique-token-plugin-pair'>;
}
declare module '@uber/typed-rpc-cli/src/graphql-to-flow/index.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/graphql-to-flow/index'>;
}
declare module '@uber/typed-rpc-cli/src/graphql-to-flow/index.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/graphql-to-flow/index.test'>;
}
declare module '@uber/typed-rpc-cli/src/index.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/index'>;
}
declare module '@uber/typed-rpc-cli/src/link.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/link'>;
}
declare module '@uber/typed-rpc-cli/src/load-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/load-config'>;
}
declare module '@uber/typed-rpc-cli/src/main.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/main'>;
}
declare module '@uber/typed-rpc-cli/src/mod-flow.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/mod-flow'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/codegen-link-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/codegen-link-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/codegen-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/codegen-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate-graphql-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate-graphql-code'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate-register-graphql-links.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate-register-graphql-links'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate-resolvers.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate-resolvers'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate-rpc-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate-rpc-code'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate-schemas.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate-schemas'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/generate.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/generate'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/get-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/get-config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/get-filtered-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/get-filtered-config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/get-paths.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/get-paths'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/get-rpc-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/get-rpc-code'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/idl.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/idl'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/index.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/index'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/install-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/install-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/link.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/link'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/lint-projects.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/lint-projects'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/merge-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/merge-config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/babel.config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/babel.config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/enzyme.config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/enzyme.config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/globals.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/globals'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme_v3.x.x.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme_v3.x.x'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme-adapter-react-16_vx.x.x.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/enzyme-adapter-react-16_vx.x.x'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/jest_v23.x.x.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/flow-typed/npm/jest_v23.x.x'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/plugin-template/src/test-utils/graphql-test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/plugin-template/src/test-utils/graphql-test'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/register-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/register-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/scaffold-plugin.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/scaffold-plugin'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/update-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/update-config'>;
}
declare module '@uber/typed-rpc-cli/src/monorepo/update-link-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/monorepo/update-link-config'>;
}
declare module '@uber/typed-rpc-cli/src/parse-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/parse-config'>;
}
declare module '@uber/typed-rpc-cli/src/proto-to-flow/__tests__/index.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/proto-to-flow/__tests__/index.test'>;
}
declare module '@uber/typed-rpc-cli/src/proto-to-flow/index.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/proto-to-flow/index'>;
}
declare module '@uber/typed-rpc-cli/src/remove-unused-imports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/remove-unused-imports'>;
}
declare module '@uber/typed-rpc-cli/src/replace-extension.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/replace-extension'>;
}
declare module '@uber/typed-rpc-cli/src/test-utils/run-test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/test-utils/run-test'>;
}
declare module '@uber/typed-rpc-cli/src/to-cjs.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/to-cjs'>;
}
declare module '@uber/typed-rpc-cli/src/update-flow.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/update-flow'>;
}
declare module '@uber/typed-rpc-cli/src/update-main-schema.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/update-main-schema'>;
}
