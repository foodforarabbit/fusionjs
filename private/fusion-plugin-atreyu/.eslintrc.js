module.exports = {
  extends: [
    require.resolve('eslint-config-uber-universal-stage-3')
  ]
};
