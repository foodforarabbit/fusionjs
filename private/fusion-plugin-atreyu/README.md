# Atreyu plugin for fusion

This is a fusion plugin for [Atreyu](https://code.uberinternal.com/diffusion/WEATREY/browse/master/#what-is-atreyu)

Atreyu is data querying/aggregation library similar to GraphQL, but designed to be used with Uber's micro-service infrastructure.

```js
// src/main.js
import App from '@fusion/react';
import Atreyu from '@uber/fusion-plugin-atreyu';
import {config} from './atreyu/config';

export default () => {
  const app = new App();
  const Atreyu = app.plugin(AtreyuPlugin, {config, M3, Logger, Tracer, Galileo})
  // atreyu client
  const atreyu = Atreyu.of();
}

// src/atreyu/config.js
export default {
  config: {
    serviceNames: ["safari"],
  },
}

#### Plugin registration

```
const Services = app.plugin(Atreyu, {config, M3, Logger, Tracer, Galileo, TChannel})
```

- `config: {serviceNames, services}` - an Atreyu config (see https://code.uberinternal.com/diffusion/WEATREY/)
- `M3` - m3 plugin
- `Logger` - logger plugin
- `Tracer` - tracer plugin
- `Galileo` - galileo plugin
- `TChannel` - tchannel plugin
