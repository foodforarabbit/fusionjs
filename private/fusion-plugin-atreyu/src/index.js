module.exports = grapheneAtreyu;

function grapheneAtreyu() {}
