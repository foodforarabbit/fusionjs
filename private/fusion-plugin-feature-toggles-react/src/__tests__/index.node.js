// @flow
/* eslint-env node */

import * as React from 'react';
import {JSDOM} from 'jsdom';

import {getService, getSimulator} from 'fusion-test-utils';
import App, {useService, FusionContext} from 'fusion-react';
import type {Context} from 'fusion-core';

import {AtreyuToken} from '@uber/fusion-plugin-atreyu';

// import FeatureTogglesPlugin from '../plugin.js';
import FeatureTogglesPlugin from '@uber/fusion-plugin-feature-toggles';
import {
  FeatureTogglesToken,
  FeatureTogglesClientToken,
  FeatureTogglesTogglesConfigToken,
} from '../index.js';
import type {ToggleDetailsType, IFeatureTogglesClient} from '../index.js';

type AtreyuType = any;

/* Mock(s) */
const mockContextFactory: () => Context = () =>
  ({
    // necessary to handle ctx.memoized call
    memoized: new Map(),
  }: any);
const mockAtreyuFactory: () => AtreyuType = () => ({}: any);
type MockDataType = {[string]: ToggleDetailsType};
const mockClientFactory: (
  data?: MockDataType
) => Class<IFeatureTogglesClient> = data => {
  class MockClient implements IFeatureTogglesClient {
    constructor(ctx: Context): IFeatureTogglesClient {
      return this;
    }
    async load(): Promise<void> {}
    get(toggleName: string): ToggleDetailsType {
      if (!data || !data[toggleName]) return {enabled: false};
      return data[toggleName];
    }
  }
  return MockClient;
};

/* App Creator(s) */
const appCreator = (data?: MockDataType, root?: React.Element<*>) => () => {
  const app = new App(root || React.createElement('div'));

  app.register(FeatureTogglesToken, FeatureTogglesPlugin);
  app.register(AtreyuToken, mockAtreyuFactory());
  app.register(FeatureTogglesClientToken, mockClientFactory(data));
  app.register(FeatureTogglesTogglesConfigToken, ['weatherToggle']);

  return app;
};

test('simple service sanity check - toggle on/off with mocked dependencies', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    toggleOn: {
      enabled: true,
    },
    toggleOff: {
      enabled: false,
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const onResult = await instance.get('toggleOn');
  expect(onResult).not.toBeNull();
  if (!onResult) throw new Error('onResult is null!'); // necessary to appease Flow
  expect(onResult).toHaveProperty('enabled');
  expect(onResult.enabled).toBeTruthy();

  const offResult = await instance.get('toggleOff');
  expect(offResult).not.toBeNull();
  if (!offResult) throw new Error('offResult is null!'); // necessary to appease Flow
  expect(offResult).toHaveProperty('enabled');
  expect(offResult.enabled).not.toBeTruthy();

  const missingResult = await instance.get('missingToggle');
  expect(missingResult).not.toBeNull();
  if (!missingResult) throw new Error('missingResult is null!'); // necessary to appease Flow
  expect(missingResult).toHaveProperty('enabled');
  expect(missingResult.enabled).not.toBeTruthy();
});

test('test service metadata with mocked dependencies', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const result = await instance.get('weatherToggle');
  expect(result).not.toBeNull();
  if (!result) throw new Error('Result is null!'); // necessary to appease Flow
  expect(result).toHaveProperty('metadata');
  const metadata = result.metadata;
  expect(metadata).not.toBeNull();
  if (!metadata) throw new Error('Metadata is null!'); // necessary to appease Flow
  expect(metadata).toHaveProperty('isItSunnyToday');
  expect(metadata.isItSunnyToday).toBeTruthy();
  expect(metadata).toHaveProperty('areThereWaves');
  expect(metadata.areThereWaves).not.toBeTruthy();
});

test('test ssr', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const result = await instance.get('weatherToggle');
  expect(result).not.toBeNull();
  if (!result) throw new Error('Result is null!'); // necessary to appease Flow
  expect(result).toHaveProperty('metadata');
  const metadata = result.metadata;
  expect(metadata).not.toBeNull();
  if (!metadata) throw new Error('Metadata is null!'); // necessary to appease Flow
  expect(metadata).toHaveProperty('isItSunnyToday');
  expect(metadata.isItSunnyToday).toBeTruthy();
  expect(metadata).toHaveProperty('areThereWaves');
  expect(metadata.areThereWaves).not.toBeTruthy();
});

test('useService usage', async () => {
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };

  const ComponentUsingService = () => {
    const featureToggles = useService(FeatureTogglesToken);
    const context = React.useContext(FusionContext);
    const weatherToggle = featureToggles.from(context).get('weatherToggle');

    expect(weatherToggle).not.toBeNull();
    expect(weatherToggle).toBe(mockToggles.weatherToggle);

    return <>Enabled: {weatherToggle.enabled}</>;
  };

  const app = appCreator(
    mockToggles,
    React.createElement(ComponentUsingService)
  )();
  const result = await getSimulator(app).render('/');

  const dom = new JSDOM(result.body);
  const body = dom.window.document.querySelector('body');
  expect(body).toBeTruthy();
  expect(body).toHaveProperty('outerHTML');
  expect(body.outerHTML).toMatchSnapshot('simple useService usage');
});
