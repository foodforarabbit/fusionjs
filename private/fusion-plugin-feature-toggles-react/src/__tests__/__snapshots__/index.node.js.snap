// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useService usage: simple useService usage 1`] = `
"<body><div id=\\"root\\">Enabled: </div>
          <script type=\\"application/json\\" id=\\"__FEATURE_TOGGLES__\\">
            {\\\\u0022data\\\\u0022:{\\\\u0022weatherToggle\\\\u0022:{\\\\u0022enabled\\\\u0022:true,\\\\u0022metadata\\\\u0022:{\\\\u0022isItSunnyToday\\\\u0022:true,\\\\u0022areThereWaves\\\\u0022:false}}}}
          </script>
        </body>"
`;
